src/directives/view.js

```diff
@@ -257,8 +259,27 @@ class routeViewDirective {
         if (component.controller) {
           const locals = _.merge(dependencies, {$scope: viewScope, $element: element.children().eq(0)});
 
-          const controller = $controller(component.controller, locals);
-          locals.$scope[component.controllerAs] = controller;
+          try {
+            const controller = $controller(component.controller, locals);
+            locals.$scope[component.controllerAs] = controller;
+          }
+          catch (error) {
+            let errorMessage;
+
+            try {
+              if (_.isObject(error)) {
+                errorMessage = JSON.stringify(error);
+              } else {
+                errorMessage = error;
+              }
+
+            } catch (jsonError) {
```

Comment:
we lose the original error here, what about:
`errorMessage = 'Failed to serialize error object for logging: ${error}';`